src/components/GameItem.js

```diff
@@ -4,23 +4,59 @@ import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
 class GameItem extends Component {
-  propTypes = {
+  // this is so we can get setstate right?
+  constructor() {
+    super();
+    this.state = {
+      isClicked: false,
+    }
+    // could I just set this to isClicked to false by adding a line in app/gameitem?
+  }
+
+  static propTypes = {
     height: PropTypes.number.isRequired,
     layer: PropTypes.number.isRequired,
   }
 
+  onClicked = () => {
+    // console.log("clicked it");
+    this.setState({
+      isClicked: true,
+    })
+    if (this.props.type === 'litter')
+     this.props.clicked();
+  }
+
+
+
   render() {
     const itemStyle = {
       bottom: `${this.props.height}px`, // use props.height to offset from the bottom of screen
       zIndex: this.props.layer, // use props.layer to set z-index, so we display ontop of background
     };
 
     // Update this to select the correct icon for each item
-    const icon = ItemIcons.rock;
+    // so this makes it all rocks right now
+    // const icon = ItemIcons.rock;
+
+    const icon = ItemIcons[this.props.type];
+    // so its taking in a type in the form of a prop from App, when it gives a component? 
+    // and then we are storing that type name, and then finding the itemicon for it. (although why does it need to be in [] and not .)
```

Comment:
square bracket syntax lets us pass in variables/syntaxes like `this.props.type` to access the `ItemIcons` object. If you used `.`, JavaScript would read it weirdly, like `ItemIcons.this.props.type` would be read as "What's the value of `this` on `ItemIcons`? Then... what's the value of `props` on `ItemIcons.this`?" etc.